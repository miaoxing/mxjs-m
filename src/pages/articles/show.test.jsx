import {render, waitFor} from '@testing-library/react';
import $, {Ret} from 'miaoxing';
import {createPromise, bootstrap, setUrl, resetUrl} from '@mxjs/test';
import Taro from '@tarojs/taro';
import Show from './show';

bootstrap();

describe('Show', () => {
  beforeEach(() => {
    setUrl('/articles/show?id=1');
  });

  afterEach(() => {
    resetUrl();
  });

  test('basic', async () => {
    const promise = createPromise();

    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: {
            title: 'title',
            updatedAt: '2021-01-01 01:01:01',
            detail: {
              content: 'content',
            },
          },
        }),
      }));

    Taro.setBackgroundColor = jest.fn();

    const {getByText} = render(<Show />);

    await waitFor(() => {
      expect(getByText('content')).not.toBeNull();
    });

    expect(getByText('title')).not.toBeNull();
    expect(getByText('2021-01-01')).not.toBeNull();

    expect($.http).toMatchSnapshot();
  });
});
