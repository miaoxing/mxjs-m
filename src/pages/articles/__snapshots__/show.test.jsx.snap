// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show basic 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/articles/1",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;
